sdk/src/assertions/actions.rs

```diff
@@ -368,26 +398,51 @@ impl Action {
         key: S,
         value: T,
     ) -> Result<Self> {
-        let value_bytes = serde_cbor::ser::to_vec(&value)?;
-        let value = serde_cbor::from_slice(&value_bytes)?;
+        let value = serde_cbor::value::to_value(value)?;
 
-        self.parameters = Some(match self.parameters {
-            Some(mut parameters) => {
-                parameters.insert(key.into(), value);
-                parameters
-            }
+        let parameters = match &mut self.parameters {
+            Some(parameters) => parameters,
             None => {
-                let mut p = HashMap::new();
-                p.insert(key.into(), value);
-                p
+                self.parameters = Some(ParametersMap::default());
+                #[allow(clippy::unwrap_used)]
+                self.parameters.as_mut().unwrap()
```

Comment:
One method I've used is this:
let parameters = self.parameters.take().or_else(ParametersMap::default());

parameters.foo = ....

self.parameters = Some(parameters)